core/arch/arm/kernel/thread_spmc.c

```diff
@@ -633,6 +633,19 @@ static int add_mem_share_frag(struct mem_frag_state *s, void *buf, size_t flen)
 	return rc;
 }
 
+static bool is_sp_share(void *buf)
+{
+	struct ffa_mem_transaction *input_descr = NULL;
+
+	if (!IS_ENABLED(CFG_SECURE_PARTITION))
+		return false;
+
+	input_descr = (struct ffa_mem_transaction *)buf;
+
+	return input_descr->mem_access_array[0].access_perm.endpoint_id !=
```

Comment:
We must read `input_descr->mem_access_array[0].access_perm.endpoint_id` only once or it may change between each check. Think time of check, time of use atack.

Reply:
Done